test/Microsoft.AspNetCore.Owin.Tests/OwinExtensionTests.cs

```diff
@@ -63,25 +67,26 @@ public void OwinDefaultNoServices()
             {
                 builderExecuted = true;
                 serviceProvider = applicationBuilder.ApplicationServices;
-                applicationBuilder.Run(async context =>
+                applicationBuilder.Run(context =>
                 {
                     applicationExecuted = true;
                     fakeService = context.RequestServices.GetService<FakeService>();
+                    return Task.FromResult(0);
                 });
-            });
+            },
+            expectedServiceProvider);
```

Comment:
> Still need to revert this line

Why? Added `OwinDefaultNullServiceProvider()` scenario covering that case since it's different from `OwinDefaultNoServices()`.